### PR TITLE
chore: add validation to input fields such as url, min and max length…

### PR DESCRIPTION
**Pull request by MaiaS**
## WHAT IT DOES:

_Description of the pull request, what changed_

Adds better validation to url, input, textarea.
Adds min/max length to textArea.
Adds min/max to word length.

Improves prompts by calculating tokens based on word limit and also adding an adjective modifier to try to get GPT to generate more and less verbose content.

### CODE REVIEW

File: src/utils/generatePrompt.ts [hunk 1]
```diff
@@ -73,3 +81,13 @@ export function generatePromptSongSSE(text: string | string[], wordLimit: number
...
+
+/**
+ * Depending on the length of the input, a different description will be generated to influence the summary length and detail
+ */
+export const modifier = (wordLimit: number) => {
+  if (wordLimit < 50) return "an extremely short";
+  if (wordLimit < 100) return "a short";
+  if (wordLimit < 200) return "a very detailed";
+  return "multiple paragraphs of an incredibly long, detailed, and verbose and longwinded text in the style of a new york times article";
+};
```
Comment (pratikmdhr) [inline src/utils/generatePrompt.ts:95]: 💯

File: src/components/utility-components/input/SummaryLengthSlider.tsx [hunk 2]
```diff
@@ -55,14 +57,23 @@ const SummaryLengthSlider = ({
...
           id="custom-summary-length"
           onChange={(event) => {
+            if (Number(event.target.value) < minLength) {
+              setCustomLength(`${minLength}`);
+              return;
+            }
+            if (Number(event.target.value) > maxLength) {
+              setCustomLength(`${maxLength}`);
+              return;
+            }
```
Comment (pratikmdhr) [inline src/components/utility-components/input/SummaryLengthSlider.tsx]: may be should validate only when the summarize button is pressed?

noticing a bug here, the field starts blank but once any number is entered, it changes to 10.. and not able to backspace and reenter the value..

**Review by pratikmdhr — CHANGES_REQUESTED** [review]
Apart from that bug for custom summary length input field, its all looking good!

File: src/components/utility-components/input/SummaryLengthSlider.tsx (hunk 2 above)
Comment (MaiaS) [reply src/components/utility-components/input/SummaryLengthSlider.tsx]: Good point! Thanks for catching!

File: src/hooks/useOpenAiSSEResponse.ts [hunk 3]
```diff
@@ -104,11 +104,11 @@ const useOpenAiSSEResponse = ({
         : "";
 
     const multiplier = Math.min(wordLimit > 100 ? 2.5 : 1.3);
-    const maxTokenLimit = Math.min(Math.round(wordLimit * multiplier) + 200, 3500);
+    const maxTokenLimit = Math.min(Math.round(wordLimit * multiplier) + 600, 2000);
```
Comment (pratikmdhr) [inline src/hooks/useOpenAiSSEResponse.ts:107]: 🙏

**Review by pratikmdhr — APPROVED** [review]

**Comment by pratikmdhr** [comment]
@MaiaS I had a chance to test it out before we hit the limit, and its looking all good! thanks !!

**Pull request merged**